reconcile/aus/base.py

```diff
@@ -771,6 +856,16 @@ def calculate_diff(
     now = datetime.utcnow()
     gates = get_version_gates(ocm_api)
     for spec in desired_state.specs:
+        # Upgrading node pools, only required for Hypershift clusters
+        if spec.cluster.is_rosa_hypershift():
```

Comment:
i think this nodepools upgrades block might need to be extracted in its own for loop: when a control plane upgrade happens, we should keep the mutex (so not allow an other control plane to take it) and run nodepool upgrades for this cluster.

Reply:
Makes sense, once node pool upgrades add mutexes this should be straightforward! Will refactor this